src/service.rs

```diff
@@ -273,14 +305,22 @@ where
 
 impl<A> Service<A>
 where
-    A: 'static + Authorizer + Send,
+    A: 'static + Authorizer<Prompt = Prompt> + Send + Sync,
+    A::Message: Send,
+    A::Error: Send,
 {
     /// Builds a new [`Service`] from `config` and `authorizer`.
     #[inline]
     pub fn build(config: Config, authorizer: A) -> Self {
+        // FIXME: This is not the best strategy for choosing which URL we want.
+        #[cfg(debug_assertions)]
+        let url = config.dev_origin_url.as_str();
+        #[cfg(not(debug_assertions))]
+        let url = config.prod_origin_url.as_str();
```

Comment:
Do you have anything against us allowing both CORS urls always? It is a bit inconvenient to have to manage separate prod and dev builds.

Reply:
I don't want the `localhost` allowed on the production build. That's a super easy way to spoof the frontend with a trojan and we should also not test some production parts with non-production parts. I just wish there was a better way to structure this instead of relying on `debug_assertions` which is a bit sketchy but technically safe.